src/test/java/seedu/address/storage/JsonAdaptedLessonTest.java

```diff
@@ -0,0 +1,108 @@
+package seedu.address.storage;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static seedu.address.storage.JsonAdaptedLesson.END_TIME_FIELD;
+import static seedu.address.storage.JsonAdaptedLesson.INVALID_FIELD_MESSAGE_FORMAT;
+import static seedu.address.storage.JsonAdaptedLesson.MISSING_FIELD_MESSAGE_FORMAT;
+import static seedu.address.storage.JsonAdaptedLesson.START_TIME_FIELD;
+import static seedu.address.testutil.Assert.assertThrows;
+import static seedu.address.testutil.TypicalLesson.LESSON_WED_2_TO_4;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.commons.exceptions.IllegalValueException;
+import seedu.address.model.lesson.Day;
+import seedu.address.model.lesson.NumberOfOccurrences;
+import seedu.address.model.lesson.Venue;
+
+public class JsonAdaptedLessonTest {
+
+    private static final String INVALID_TIME = "1400";
+    private static final String INVALID_DAY = "wed";
+    private static final int INVALID_NUMBER_OF_OCCURRENCES = NumberOfOccurrences.UPPER_BOUND + 1;
+    private static final String INVALID_VENUE = "!@#$%^&*";
+
+    private static final String VALID_START_TIME = LESSON_WED_2_TO_4.getStartTime().toString();
+    private static final String VALID_END_TIME = LESSON_WED_2_TO_4.getEndTime().toString();
+    private static final String VALID_DAY = LESSON_WED_2_TO_4.getDay().toString();
+    private static final int VALID_NUMBER_OF_OCCURRENCES = LESSON_WED_2_TO_4.getNumberOfOccurrences().value;
+    private static final String VALID_VENUE = LESSON_WED_2_TO_4.getVenue().toString();
+
+    @Test
+    public void toModelType_validLessonDetails_returnsLesson() throws Exception {
+        JsonAdaptedLesson lesson = new JsonAdaptedLesson(LESSON_WED_2_TO_4);
+        assertEquals(LESSON_WED_2_TO_4, lesson.toModelType());
+    }
```

Comment:
Could we test both versions of `JsonAdaptedLesson` creation? That is have another test case that creates the `JsonAdaptedLesson` solely using the individual details of the `Lesson` rather than the object itself.

Reply:
Have included similar tests for `JsonAdaptedModuleClassTest` and `JsonAdaptedStudentTest`.

Reply:
Nice thanks!